How many passangers servived from 1st the titanic?

I think my study is retrospective. A retrospective study looks back at events that have already happened. The Titanic data set is historical; it contains data collected after the sinking of the Titanic. You are analyzing this past data to look for patterns, so this qualifies as a retrospective observational study.




A bias that can be produced is that not accounting for all people, because it's only ~800 people, not the full population of the Titanic.

In [10]:
!pip install pandas

Defaulting to user installation because normal site-packages is not writeable
Looking in links: /usr/share/pip-wheels


In [11]:
import pandas as pd

In [12]:
pd.read_csv("titanic.csv")

survived  pclass                                               name  \
0           0       3                            Braund, Mr. Owen Harris   
1           1       1  Cumings, Mrs. John Bradley (Florence Briggs Th...   
2           1       3                             Heikkinen, Miss. Laina   
3           1       1       Futrelle, Mrs. Jacques Heath (Lily May Peel)   
4           0       3                           Allen, Mr. William Henry   
..        ...     ...                                                ...   
709         0       3               Rice, Mrs. William (Margaret Norton)   
710         0       2                              Montvila, Rev. Juozas   
711         1       1                       Graham, Miss. Margaret Edith   
712         1       1                              Behr, Mr. Karl Howell   
713         0       3                                Dooley, Mr. Patrick   

        sex   age     fare  sibsp  parch  
0      male  22.0   7.2500      1      0  
1    female  38.0  71.2833      1      0  
2    female  26.0   7.9250      0      0  
3    female  35.0  53.1000      1      0  
4      male  35.0   8.0500      0      0  
..      ...   ...      ...    ...    ...  
709  female  39.0  29.1250      0      5  
710    male  27.0  13.0000      0      0  
711  female  19.0  30.0000      0      0  
712    male  26.0  30.0000      0      0  
713    male  32.0   7.7500      0      0  

[714 rows x 8 columns]

In [23]:
num_samples = len("titanic.csv")
print(num_samples)

11


In [25]:
print("titanic.csv" [6])

c


In [26]:
import pandas as pd

# Load the titanic dataset from the CSV file
titanic_data = pd.read_csv("titanic.csv")

print("Individuals who embarked from Queenstown (Q):")
for index, person in titanic_data.iterrows():
    if person.get("embarked") == "Q":  # Using get() method which works with pandas Series
        # Print the name or "Unknown Name" if name is not available
        print(person.get("name", "Unknown Name"))



Individuals who embarked from Queenstown (Q):


In [37]:
{
  "class": "First Class",
  "total_passengers": 325,
  "survived": 202,
  "survival_rate": "62%"
}


{'class': 'First Class',
 'total_passengers': 325,
 'survived': 202,
 'survival_rate': '62%'}

In [44]:
import pandas as pd

# Load without assuming headers
df = pd.read_csv("titanic.csv", header=None)

# Show first 10 rows so we can see what the data looks like
print(df.head(10))

# Show how many columns pandas detected
print("Number of columns:", df.shape[1])


          0       1                                                  2  \
0  survived  pclass                                               name   
1         0       3                            Braund, Mr. Owen Harris   
2         1       1  Cumings, Mrs. John Bradley (Florence Briggs Th...   
3         1       3                             Heikkinen, Miss. Laina   
4         1       1       Futrelle, Mrs. Jacques Heath (Lily May Peel)   
5         0       3                           Allen, Mr. William Henry   
6         0       1                            McCarthy, Mr. Timothy J   
7         0       3                     Palsson, Master. Gosta Leonard   
8         1       3  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)   
9         1       2                Nasser, Mrs. Nicholas (Adele Achem)   

        3     4        5      6      7  
0     sex   age     fare  sibsp  parch  
1    male  22.0     7.25      1      0  
2  female  38.0  71.2833      1      0  
3  female  26.0    7.

In [2]:





titanic_data = pd.read_csv("titanic.csv")

# Display total number of samples (rows)
num_samples = len(titanic_data)
print("Total number of passengers:", num_samples)

print("7th passenger:", titanic_data.loc[6, 'name'])


print("\nIndividuals who embarked from Queenstown (Q):")
for index, person in titanic_data.iterrows():
    if person.get("embarked") == "Q":
        print(person.get("name", "Unknown Name"))


first_class = titanic_data[titanic_data["pclass"] == 1]
total_fc = len(first_class)
survived_fc = len(first_class[first_class["survived"] == 1])
survival_rate_fc = (survived_fc / total_fc) * 100

print(f"\nFirst Class Survival Rate: {survival_rate_fc:.2f}%")


Total number of passengers: 714
7th passenger: Palsson, Master. Gosta Leonard

Individuals who embarked from Queenstown (Q):

First Class Survival Rate: 65.59%



Out of all the people in the first class 65.59% of he passangers servived. this number is much bigger then the other levels  